docs/modules/deploy/pages/deploying-in-kubernetes.adoc

```diff
@@ -50,3 +59,401 @@ Since all the Operators mentioned above are generated from Helm Charts, their fu
+        +zKqa06JKKrz3Qony97VdWR/NMpRYXouSF0owDv9BIoLTC682wlQtNB1c4pTVW7a
+        AikoNtyNIT8gtA5w0MyjFrbNslUblXvuo0HIeSmJREUmT7BC3VaKgkg64mVdf0DJ
+        NyrcL+qyCs1m03mi12hgzI72O3qgEtP91tu/oCUdOh39u13TB0fj5tgWURMFgkxo
+        T0xiNfPueV3pe8uYxBntzFn/74ibiizLRP6d/hsuRdS7IA+bvRLKG/paYwyZuMFb
+        BDA+kXXAIkOvCpIQCkAKMpyyDz9lBVCtl3eRSAJQLBefAgMBAAGjIzAhMA4GA1Ud
+        DwEB/wQEAwICBDAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQBP
+        TBRY1IbkFJuboMKLW9tdpIzW7hf2qsTLOhtlaJbMXrWaXCTrl8qUgBUZ1sWAW9Uk
+        qETwRoCMl1Ht7PhbnXEGDNt3Sw3Y3feR4PsffhcgWH0BK8pZVY0Q1zbZ6dVNbU82
+        EUrrcnV0uiB/JFsJ3rg8qJurutro3uIzAhb9ixYRqYnXUR4q0bxahO04iSUHvtYQ
+        JmWp1GCb/ny9MyeTkwh2Q+WIQBHsX4LfrKjPwJd6qZME7BmwryYBTkGa0FinmhRg
+        SdSPEQKmuXmghPU5GLudiI2ooOaqOXIjVPfM/cw4uU9FCGM49qufccOOt6utk0SM
+        DwupAKLLiaYs47a8JgUa
+        -----END CERTIFICATE-----
+```
+
+NOTE: Hazelcast Client outside Kubernetes cluster works only in the **Kubernetes API** mode (it does not work in the **DNS Lookup** mode).
+
+NOTE: If you use Minikube, you need to execute `minikube tunnel` in order to get LoadBalancer External IPs assigned.
+
+== Rolling Upgrade and Scaling
+
+Hazelcast cluster is easily scalable within Kubernetes. You can use the standard `kubectl scale` command to change the cluster size. The same applies the rolling upgrade procedure, you can depend on the standard Kubernetes behavior and just update the new version to your Deployment/StatefulSet configurations.
+
+Note however that, by default, Hazelcast does not shutdown gracefully. It means that if you suddenly terminate more than your `backup-count` property (1 by default), you may lose the cluster data. To prevent that from happening, set the following properties:
+
+- `terminationGracePeriodSeconds`:  in your StatefulSet (or Deployment) configuration; the value should be high enough to cover the data migration process
+- `-Dhazelcast.shutdownhook.policy=GRACEFUL`: in the JVM parameters
+- `-Dhazelcast.graceful.shutdown.max.wait`: in the JVM parameters; the value should be high enough to cover the data migration process
+
+Additionally:
+- If you use Deployment (not StatefulSet), you need to set your strategy to link:https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#updating-a-deployment[RollingUpdate] and ensure Pods are updated one by one.
+- If you upgrade by the minor version, e.g., `3.11.4 => 3.12` (Enterprise feature), you need to set the following JVM property `-Dhazelcast.cluster.version.auto.upgrade.enabled=true` to make sure the cluster version updates automatically.
+
+All these features are already included in <<helm-chart, Hazelcast Helm Charts>>.
+
+== CP Subsystem
+
+Hazelcast CP Subsystem can be used safely in Kubernetes only if CP Subsystem Persistence is enabled (Enterprise Feature). Otherwise, a CP Subsystem Group may not recover after scaling the cluster or performing the rolling upgrade operation.
```

Comment:
Maybe giving a cross-reference to cp-subsystem:configuration.adoc#cp-subsystem-configuration would be nice which is on how to enable the CP subsystem persistence.